File: backend/app/schemas/signals.py
from pydantic import BaseModel


class SignalDetailOut(BaseModel):
    name: str
    score: float
    weight: float
    description: str


class RiskContext(BaseModel):
    stop_loss: float | None
    target_price: float | None
    risk_reward: float | None
    position_size: float
    position_pct: float


class ScanResult(BaseModel):
    symbol: str
    current_price: float
    direction: str
    conviction: str
    composite_score: float
    confidence: int
    signals: list[SignalDetailOut]
    risk: RiskContext

File: frontend/src/types/signals.ts
export interface SignalDetail {
  name: string;
  score: number;
  weight: number;
  description: string;
}

export interface RiskContext {
  stop_loss: number | null;
  target_price: number | null;
  risk_reward: number | null;
  position_size: number;
  position_pct: number;
}

export interface ScanResult {
  symbol: string;
  current_price: number;
  direction: "buy" | "sell" | "hold";
  conviction: "low" | "medium" | "high";
  composite_score: number;
  confidence: number;
  signals: SignalDetail[];
  risk: RiskContext;
}
